Subject: create sysfs device symlinks for snd-aoa
From: olh@suse.de
References: 106294
Patch-mainline: yes

make hal happy, provide device symlinks for sound devices
make security happy, no need to put the default user in group audio
make hwinfo happy, provide module symlink in case it cares about this


---
 sound/aoa/aoa.h                           |    2 +-
 sound/aoa/core/snd-aoa-alsa.c             |    3 ++-
 sound/aoa/core/snd-aoa-alsa.h             |    2 +-
 sound/aoa/core/snd-aoa-core.c             |    4 ++--
 sound/aoa/fabrics/snd-aoa-fabric-layout.c |    5 ++++-
 5 files changed, 10 insertions(+), 6 deletions(-)

Index: linux-2.6/sound/aoa/aoa.h
===================================================================
--- linux-2.6.orig/sound/aoa/aoa.h
+++ linux-2.6/sound/aoa/aoa.h
@@ -99,7 +99,7 @@ struct aoa_fabric {
  * that are not assigned yet are passed to the fabric
  * again for reconsideration. */
 extern int
-aoa_fabric_register(struct aoa_fabric *fabric);
+aoa_fabric_register(struct aoa_fabric *fabric, struct device *dev);
 
 /* it is vital to call this when the fabric exits!
  * When calling, the remove_codec will be called
Index: linux-2.6/sound/aoa/fabrics/snd-aoa-fabric-layout.c
===================================================================
--- linux-2.6.orig/sound/aoa/fabrics/snd-aoa-fabric-layout.c
+++ linux-2.6/sound/aoa/fabrics/snd-aoa-fabric-layout.c
@@ -1014,7 +1014,7 @@ static int aoa_fabric_layout_probe(struc
 
 	ldev->gpio.methods->init(&ldev->gpio);
 
-	err = aoa_fabric_register(&layout_fabric);
+	err = aoa_fabric_register(&layout_fabric, &sdev->ofdev.dev);
 	if (err && err != -EALREADY) {
 		printk(KERN_INFO "snd-aoa-fabric-layout: can't use,"
 				 " another fabric is active!\n");
@@ -1107,6 +1107,9 @@ static struct soundbus_driver aoa_soundb
 	.suspend = aoa_fabric_layout_suspend,
 	.resume = aoa_fabric_layout_resume,
 #endif
+	.driver = {
+		.owner = THIS_MODULE,
+	}
 };
 
 static int __init aoa_fabric_layout_init(void)
Index: linux-2.6/sound/aoa/core/snd-aoa-alsa.c
===================================================================
--- linux-2.6.orig/sound/aoa/core/snd-aoa-alsa.c
+++ linux-2.6/sound/aoa/core/snd-aoa-alsa.c
@@ -14,7 +14,7 @@ MODULE_PARM_DESC(index, "index for AOA s
 
 static struct aoa_card *aoa_card;
 
-int aoa_alsa_init(char *name, struct module *mod)
+int aoa_alsa_init(char *name, struct module *mod, struct device *dev)
 {
 	struct snd_card *alsa_card;
 	int err;
@@ -28,6 +28,7 @@ int aoa_alsa_init(char *name, struct mod
 		return -ENOMEM;
 	aoa_card = alsa_card->private_data;
 	aoa_card->alsa_card = alsa_card;
+	alsa_card->dev = dev;
 	strlcpy(alsa_card->driver, "AppleOnbdAudio", sizeof(alsa_card->driver));
 	strlcpy(alsa_card->shortname, name, sizeof(alsa_card->shortname));
 	strlcpy(alsa_card->longname, name, sizeof(alsa_card->longname));
Index: linux-2.6/sound/aoa/core/snd-aoa-alsa.h
===================================================================
--- linux-2.6.orig/sound/aoa/core/snd-aoa-alsa.h
+++ linux-2.6/sound/aoa/core/snd-aoa-alsa.h
@@ -10,7 +10,7 @@
 #define __SND_AOA_ALSA_H
 #include "../aoa.h"
 
-extern int aoa_alsa_init(char *name, struct module *mod);
+extern int aoa_alsa_init(char *name, struct module *mod, struct device *dev);
 extern void aoa_alsa_cleanup(void);
 
 #endif /* __SND_AOA_ALSA_H */
Index: linux-2.6/sound/aoa/core/snd-aoa-core.c
===================================================================
--- linux-2.6.orig/sound/aoa/core/snd-aoa-core.c
+++ linux-2.6/sound/aoa/core/snd-aoa-core.c
@@ -82,7 +82,7 @@ void aoa_codec_unregister(struct aoa_cod
 }
 EXPORT_SYMBOL_GPL(aoa_codec_unregister);
 
-int aoa_fabric_register(struct aoa_fabric *new_fabric)
+int aoa_fabric_register(struct aoa_fabric *new_fabric, struct device *dev)
 {
 	struct aoa_codec *c;
 	int err;
@@ -98,7 +98,7 @@ int aoa_fabric_register(struct aoa_fabri
 	if (!new_fabric)
 		return -EINVAL;
 
-	err = aoa_alsa_init(new_fabric->name, new_fabric->owner);
+	err = aoa_alsa_init(new_fabric->name, new_fabric->owner, dev);
 	if (err)
 		return err;
 
